thentos-adhocracy/tests/Thentos/Adhocracy3/Backend/Api/SimpleSpec.hs

```diff
@@ -91,6 +95,40 @@ spec =
                      (Aeson.eitherDecode userdata :: Either String A3UserWithPass)
                      `shouldBe` Left "Illegal whitespace sequence in user name: \" Anna  Toll\""
 
+        describe "login" . before setupTestBackend . after teardownTestBackend $
+            it "rejects bad credentials mimicking A3" $
+                \bts -> runTestBackend bts $ do
+
+                    let reqBody = encodePretty . object $
+                            [ "name"     .= String "Anna Müller"
+                            , "password" .= String "this-is-wrong"
+                            ]
+
+                    rsp <- srequest $ makeSRequest "POST" "login_username" [] reqBody
+                    shouldBeErr400WithCustomMessage (simpleStatus rsp) (simpleBody rsp)
+                        "\"User doesn't exist or password is wrong\""
+
+        describe "activate_account" . before setupTestBackend . after teardownTestBackend $
+            it "rejects bad path mimicking A3" $
+                \bts -> runTestBackend bts $ do
+
+                    let reqBody = encodePretty . object $
+                            [ "path" .= String "/activate/no-such-path" ]
+
+                    rsp <- srequest $ makeSRequest "POST" "activate_account" [] reqBody
```

Comment:
Well I thought about importing the JSON class used to generate error and check that we can convert the serialized JSON back to it, but that felt a bit like cheating. It drilling into JSON *without* a defining type class is no fun at all (I did a bit of that in other tests).

So I choose to just make sure that it *is* JSON and contains the specifically relevant bits of information (including the surrounding quotation marks added by JSON). For me that's good enough, but if you want to improve the tests. feel free to do so.